src/server/json_family.cc

```diff
@@ -692,8 +702,135 @@ OpResult<vector<OptSizeT>> OpArrTrim(const OpArgs& op_args, string_view key, str
   return vec;
 }
 
+OpResult<vector<OptLong>> OpArrIndex(const OpArgs& op_args, string_view key,
+                                     JsonExpression expression, const json& search_val,
```

Comment:
can we search for any json object? even for an object?

Reply:
No. according to Elasticache documentation, only JSON scalar is searchable.
"JSON scalar refers to values that are not objects or arrays. i.e., String, number, boolean and null are scalar values."

Reply:
please add a comment quoting this.